Automattic/wp-calypso
# Domains: Create redesigned domain settings page sidebar

#### Changes proposed in this Pull Request
Adds the sidebar elements to the new domain settings page. 
This is part of the domain management redesign project described in pcYYhz-m2-p2.

#### Preview
<img width="420" src="https://user-images.githubusercontent.com/18705930/145658879-a48c7b35-d93d-4deb-8179-d12f76a5d1a1.png" />
<img width="420" src="https://user-images.githubusercontent.com/18705930/145658883-52f3efc5-e955-4ed4-8962-10b8c6bd0a58.png" />
<img width="420" src="https://user-images.githubusercontent.com/18705930/145658890-0af8b6c1-2af9-41a4-8c0b-476444b54e91.png" />
<img width="420" src="https://user-images.githubusercontent.com/18705930/145658899-d8e0a7f1-b7f8-4ef4-9ebc-c430734f9440.png" />
<img width="420" src="https://user-images.githubusercontent.com/18705930/145658955-3396e5de-845d-41c2-aa5e-c1d4debe8780.png" />


#### Testing instructions

- Build this branch locally or open the live Calypso link
- Go to "Upgrades > Domains"
- Select one of your registered domains or domain connections
- If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Please check that:
  - The sidebar is shown as in the screenshots and as described in the redesign (hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A53742)
  - Email count works as expected;
  - All buttons redirect to the correct page;
  - Transfer card is not rendered if the domain is a transfer;
  - Delete/cancel button is not rendered if the purchase is not removable (_follows the same logic as in the [`RemovePurchase`](https://github.com/Automattic/wp-calypso/commit/3cb00c240c54273de0c372daa7afe3dff7ab946c) component_);
  - The cards' labels change according to the domain type: connection, transfer, etc);
  - Mobile view looks correct as shown in the screenshot above;
  - No other elements have been affected by the changes, and also make sure that the page without the feature flag looks the same as before.
